### PR TITLE
improved gracefulness of adding emails to user (put logic inside that co...

### PR DESCRIPTION
**Pull request by jpslav**
...mmon code, handle verifiying of existing emails), and took out a typo line from key setup

@lakshmivyas I took out that line that appeared errant from your contribution. 

@Dantemss to review.

### CODE REVIEW

File: app/routines/add_email_to_user.rb [hunk 1]
```diff
@@ -7,13 +7,28 @@ class AddEmailToUser
...
+    # If it is a brand new email address, make it and send confirmation
+    if email_address.nil?
+      email_address = EmailAddress.new(value: email_address_text)
+      email_address.user = user
+      email_address.verified = options[:already_verified] || false
+
+      email_address.save
+      transfer_errors_from(email_address, { scope: :email_address }, true)
+
+      run(SendContactInfoConfirmation, email_address)
```
Comment (Dantemss) [inline app/routines/add_email_to_user.rb]: Don't wanna send this if already verified.

File: app/routines/transfer_omniauth_data.rb [hunk 2]
```diff
@@ -11,8 +11,6 @@ def exec(data, user)
     raise Unexpected if data.provider == 'identity'
 
     run(AddEmailToUser, data.email, user, {already_verified: true}) \
```
Comment (Dantemss) [inline app/routines/transfer_omniauth_data.rb]: Remove the \

**Comment by Dantemss** [comment]
Couple of comments, otherwise looks good.

File: app/routines/add_email_to_user.rb (hunk 1 above)
Comment (jpslav) [inline app/routines/add_email_to_user.rb]: actually SendContactInfoConfirmation doesn't send if already verified.  But that means I should be able to simplify this a bit.

File: app/routines/transfer_omniauth_data.rb (hunk 2 above)
Comment (jpslav) [inline app/routines/transfer_omniauth_data.rb]: good catch

**Pull request merged**